executor/insert_test.go

```diff
@@ -321,7 +321,7 @@ func (s *testSuite3) TestInsertWrongValueForField(c *C) {
 	tk.MustExec(`drop table if exists t;`)
 	tk.MustExec(`create table t (a year);`)
 	_, err = tk.Exec(`insert into t values(2156);`)
-	c.Assert(err.Error(), Equals, `[types:1264]Out of range value for column 'a' at row 1`)
+	c.Assert(err.Error(), Equals, `[types:8033]invalid year`)
```

Comment:
why we change this test?

Reply:
I think it's more clearly to tell user the real reason about why cannot insert "2156" as year is because the "2156" is not a valid year. 
The original error message maybe a little confuse (after all "2156" is not some special cases like too big or too small..).